### ISSUE TITLE
Azure AD connection timeout on app service only

### ISSUE THREAD

**Issue by nihonjinboy85** [tediousjs/tedious#1416]
When refactoring a Node.js API to use Azure AD instead of SQL Server auth, I've successfully implemented the `azure-active-directory-password` auth locally, but when I deploy it to my web app, I receive an `ETIMEOUT` error stating `Failed to connect to [server-name].database.windows.net:1433 in 15000ms`.

### Things I have checked/tried
- Created managed identity for web app
- Confirmed existence of all env vars, both locally and within web app
- Set SQL server to "Allow Azure services and resources to access this server"
- Added AAD user and managed identity as user within DB
- Assigned AAD user managed identity `db_datareader` and `db_datawriter` roles within DB
- Assigned managed identity, but NOT AAD user, `SQL DB Contributor` role within Azure SQL Server Access Control (IAM)
- Attempt auth from web app using `azure-active-directory-password` and `azure-active-directory-msi-app-service`

### index.js
```javascript
const express = require('express');
const sql = require('mssql');

// SQL config
const config = {
  authentication: {
    type: 'azure-active-directory-msi-app-service',
    options: {
      clientId: 'f3405f2b-b35b-4533-83e7-36a3abac6945',
      msiEndpoint: process.env.MSI_ENDPOINT,
      msiSecret: process.env.MSI_SECRET
    }
  },
  server: process.env.DB_SERVER,
  database: process.env.DB_NAME,
  options: {
    encrypt: true,
    trustServerCertificate: process.env.NODE_ENV === 'development'
  },
  pool: {
    max: 10,
    min: 0,
    idleTimeoutMillis: 30000
  }
};

if (process.env.NODE_ENV === 'development') {
  config.authentication.type = 'azure-active-directory-password';
  config.authentication.options = {
    userName: process.env.DB_USERNAME,
    password: process.env.DB_PASSWORD,
    tenantId: process.env.DB_TENANT_ID
  };
}

// Express config
const app = express();
app.use(express.json());
app.use(express.urlencoded({ extended: true }));

app.get('/candidates', async (req, res) => {
  try {
    // Attempt SQL query
    const pool = await sql.connect(config);
    pool.request().query(`SELECT * FROM Candidate`, (err, result) => {
      if (err) {
        throw err;
      } else {
        res.json(result.recordset);
      }
    });
  } catch (err) {
    console.error(err);
    res.send(err.message);
  }
});

app.get('*', (req, res) => {
  res.send('Connected');
});

app.listen(process.env.PORT, () => {
  console.log(`Server listening on port ${process.env.PORT}...`);
});

```

Any assistance will be greatly appreciated. Thank you!

**Comment by mShan0** [comment]
Hi @nihonjinboy85, we are currently looking into this. 
In the meantime, does using `azure-active-directory-access-token` work if you grab the token elsewhere?

**Comment by nihonjinboy85** [comment]
Hi @mShan0 , thanks for looking into this so quickly. Great news! I was actually able to resolve this issue. It turns out that the issue was caused by my app service web app somehow using Node.js v10 instead of v16, which is what my local machine was using. Therefore, some of the features that the `mssql` package was expecting weren't available once deployed. Once I set the Node.js and NPM versions in my `package.json` and updated the web app's configuration settings, everything worked as expected.
